src/core/js/views/componentView.js

```diff
@@ -23,17 +23,58 @@ define([
             this.renderState();
         },
 
+        /**
+         * Allows components that want to use inview for completion to set that up
+         * @param {string} [inviewElementSelector] Allows to you to specify (via a selector) which DOM element to use for inview.
+         * Defaults to `'.component-inner'` if not supplied.
+         * @param {function} [callback] Allows you to specify what function is called when the component has been viewed.
+         * Defaults to `view.setCompletionStatus` if not specified
+         */
+        setupInviewListener: function(inviewElementSelector, callback) {
+            this.$inviewElement = this.$(inviewElementSelector || '.component-inner');
+            this.inviewCallback = (callback || this.setCompletionStatus);
+
+            this.$inviewElement.on('inview.componentView', this.onInview.bind(this));
+        },
+
+        removeInviewListener: function() {
+            this.$inviewElement.off('inview.componentView');
+        },
+
+        onInview: function(event, visible, visiblePartX, visiblePartY) {
+            if (!visible) return;
+
+            switch (visiblePartY) {
+				case 'top':
+					this.hasSeenTop = true;
+					break;
+				case 'bottom':
+					this.hasSeenBottom = true;
+					break;
+				case 'both':
+					this.hasSeenTop = this.hasSeenBottom = true;
+			}
```

Comment:
Indentation has gone a bit funky here.

Reply:
that's what I get for copy-pasting some of @tomgreenfield 's code ;-)